components/common/dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,211 @@
+
+const handleSelect = (value: string | number) => {
+  console.log(`Selected value: ${value}`);
+  // 여기에 원하는 액션을 추가
+};
+
+<Dropdown
+        options={options1}
+        onSelect={handleSelect}
+        type="select"
+        width={368} // 커스텀 너비
+        initialLabel="Red" // 초기 레이블 설정
+      />
+
+       <Dropdown
+        options={options2}
+        onSelect={handleSelect}
+        type="action"
+        trigger={<span className="cursor-pointer">O</span>}
+        optionClassName="text-center" // 옵션 항목의 스타일
+      />
+ */
+function Dropdown({
+  options,
+  onSelect,
+  type,
+  trigger,
+  initialLabel,
+  width = 126,
+  dropdownClassName,
+  optionClassName,
+}: DropdownProps) {
+  const [isOpen, setIsOpen] = useState(false);
+  const [selectedLabel, setSelectedLabel] = useState<string | undefined>(
+    initialLabel
+  );
+  const dropdownRef = useRef<HTMLDivElement | null>(null);
+
+  // 클릭 외부 감지
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (
+        dropdownRef.current &&
+        !dropdownRef.current.contains(event.target as Node)
+      ) {
+        setIsOpen(false);
+      }
+    };
+
+    // 전역 클릭 이벤트 리스너 등록
+    document.addEventListener('mousedown', handleClickOutside);
+
+    // 컴포넌트 언마운트 시 리스너 제거
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, []);
+
+  const handleSelect = (option: Option) => {
+    onSelect(option.value);
+    setSelectedLabel(option.label);
+    setIsOpen(false);
+  };
+
+  return (
+    <div className="relative inline-block text-left" ref={dropdownRef}>
+      {type === 'action' ? (
+        <div onClick={() => setIsOpen(!isOpen)}>{trigger}</div>
+      ) : (
+        <div
+          onClick={() => setIsOpen(!isOpen)}
+          className={`flex items-center cursor-pointer border rounded-[16px] px-4 py-2 ${dropdownClassName} text-gray-700 font-medium bg-white ring-1 ring-gray-300`}
+          style={{ width: `${width}px` }}
+        >
+          <span className={`flex-1 ${isOpen ? 'text-gray-300' : ''}`}>
+            {selectedLabel || initialLabel || 'Select'}
+          </span>
+          <span className="ml-2">
+            {isOpen ? (
+              <Image
+                width={24}
+                height={24}
+                src="/dropdown_arrow_up.png"
+                alt="위 방향 화살표"
+              />
+            ) : (
+              <Image
+                width={24}
+                height={24}
+                src="/dropdown_arrow_down.png"
+                alt="아래 방향 화살표"
+              />
+            )}
+          </span>
+        </div>
+      )}
+
+      {isOpen && (
+        <div
+          className={`absolute mt-2 rounded-[16px] bg-white ring-1 ring-gray-300 ${dropdownClassName} ${styles.fadeInDown}`}
+          style={{ width: `${width}px` }}
+        >
+          <div className="py-1">
+            {options.map((option, index) => (
+              <button
+                key={index}
```

Comment:
```suggestion
                key={option.key}
```
리액트에서는 성능과 관련된 문제로 key에 index값을 넣는 것을 피하는 것이 좋아요.